### DISCUSSION TITLE
Beta updates after v15.3.0

### DISCUSSION THREAD

**Discussion by tekezo** [pqrs-org/Karabiner-Elements#4007] [Announcements]
# v15.3.1

-   🐛 Bug Fixes
    -   Fixed an overflow issue with the mouse key movements.

**Comment by tekezo** [comment]
# v15.3.2

-   Support the following keys:
    -   consumer::ac_zoom_out
    -   consumer::ac_zoom_in
    -   generic_desktop::system_sleep

**Comment by tekezo** [comment]
# v15.3.3

-   Fixed an issue that the fn key is unintentionally tapped when use the media keys if "Use all F1, F2, etc. keys as standard function keys" is enabled.

**Comment by tekezo** [comment]
# v15.3.4

-   ✨ New Features
    -   Added `history_index` option into `open_application`.

**Comment by tekezo** [comment]
# v15.3.5

- Improve open_application history_index handling

**Comment by tekezo** [comment]
# v15.3.6

- history_index -> frontmost_application_history_index

**Comment by tekezo** [comment]
# v15.3.7

-   ✨ New Features
    -   Added `conditions` option into `to`.

**Reply by evan-liu** [reply]
Have been waiting for this for years for improving layers and tap-hold modifiers. Thank you so much for adding it.

**Comment by tekezo** [comment]
# v15.3.8

-   ⚡️ Improvements
    -   Added support for new consumer keys (Thanks to @indygriffiths):
        -   `{ "consumer_key_code": "stop" }`
        -   `{ "consumer_key_code": "bass_boost" }`
        -   `{ "consumer_key_code": "loudness" }`
        -   `{ "consumer_key_code": "bass_increment" }`
        -   `{ "consumer_key_code": "bass_decrement" }`
        -   `{ "consumer_key_code": "ac_search" }`

**Reply by LewsTherinTelescope** [reply]
Found myself lamenting my inability to remap the search key only to find it was added to beta literally yesterday, 🎉

**Comment by stoper123** [comment]
I am not able to do simple modification 
![Screenshot 2024-12-16 at 10 07 06 PM](https://github.com/user-attachments/assets/a31ff537-a830-4329-a909-b015e3fba84c)
![Uploading Screenshot 2024-12-16 at 10.07.39 PM.png…]()

**Comment by tekezo** [comment]
# v15.3.9

- Fix tests

**Comment by tekezo** [comment]
# v15.3.10

- The implicit conversion of Touch ID to the Lock key has been updated to work even when modifiers are held down

**Comment by tekezo** [comment]
# v15.3.11

- Change label in simple_modifications.json; fn -> fn (globe)

**Comment by tekezo** [comment]
# v15.3.12

-   ✨ New Features
    -   Added `XY movement multiplier` and `Wheels multiplier` to the mouse settings in Devices tab.

<img width="1100" alt="Screenshot 2024-12-31 at 8 24 17" src="https://github.com/user-attachments/assets/269feea0-91b4-4083-863d-f49f021dc8fd" />

<img width="1100" alt="Screenshot 2024-12-31 at 8 24 13" src="https://github.com/user-attachments/assets/f9e1c582-245c-4860-adae-66f5a9e2220c" />

**Comment by tekezo** [comment]
# v15.3.13

- Tweak DevicesMouseSettingsView

**Comment by tekezo** [comment]
# v15.3.14

-   ⚡️ Improvements
    -   Improved `to_delayed_action` behavior to use the modifier flags of the time the key is pressed, when sending events.

**Comment by tekezo** [comment]
# v15.3.15

- Temporarily set the initial size of the settings window.

**Comment by tekezo** [comment]
# v15.3.16

-   ✨ New Features
    -   Added `Ignore vendor events` option to device settings in Devices tab.
    -   Added `Filter useless events from specific devices` option in Expert tab.

**Comment by tekezo** [comment]
# v15.3.17

-   💥 Breaking changes
    -   The reordering of key events with the same timestamp is now disabled by default.
        It has been added as an option in the Expert tab: `Reorder same timestamp input events to prioritize modifiers`.

**Comment by tekezo** [comment]
# v15.3.18

-   ✨ New Features
    -   Support Headset buttons.

**Comment by tekezo** [comment]
# v15.3.19

- Add headset icon into DeviceSelectorView.

**Comment by tekezo** [comment]
# v15.3.20

- Restored the default behavior of enabling key event reordering for events with the same timestamp.

**Comment by tekezo** [comment]
# v15.3.22

- Added Karabiner-Updater and separated the update functionality from Settings.

**Comment by tekezo** [comment]
# v15.3.23

- Improved updater behavior to bring the application to the front before displaying a dialog.

**Comment by tekezo** [comment]
# v15.3.24

- Migrating Menu to the SwiftUI life cycle.

**Comment by tekezo** [comment]
# v15.3.25

- Migrating Settings to the SwiftUI life cycle.

**Comment by tekezo** [comment]
# v15.3.26

- Improve Settings behaivor:
    - Call openApplication on the app itself within handleGetURLEvent.

**Comment by tekezo** [comment]
# v15.3.27

- Migrating GamePadViewer to the SwiftUI life cycle.
- Migrating EventViewer to the SwiftUI life cycle.
- Migrating NotificationWindow to the SwiftUI life cycle.
- Migrating MultitouchExtension to the SwiftUI life cycle.

**Comment by tekezo** [comment]
# v15.3.28

- Migrated the C++ package manager from cget to CPM.cmake.
- Update Karabiner-DriverKit-VirtualHIDDevice. (5.0.0 -> 6.0.0)

**Comment by tekezo** [comment]
# v15.3.29

- Use pqrs::osx::workspace::open_application_by_file_path in launch_updater.
- Update Karabiner-Updater.app: Check updates in applicationDidFinishLaunching.

**Comment by tekezo** [comment]
# v15.3.30

- Set LSUIElement false in Updater.

**Comment by tekezo** [comment]
# v15.3.31

- Use @AppStorage instead of @UserDefault.

**Comment by tekezo** [comment]
# v15.3.32

- Use NavigationSplitView in EventViewer.

**Comment by tekezo** [comment]
# v15.3.33

- Prevent the updater from restarting after the update.
- Use NavigationSplitView.

**Comment by tekezo** [comment]
# v15.3.34

- Improve settings window creation in MultitouchExtension.
- Update japanese_eisuu and japanese_kana label.

**Comment by tekezo** [comment]
# v15.3.35

- Added a toggle to EventViewer for monitoring events.
- Update runs-on (macos-14 -> macos-15).
- Update system requirements to build Karabiner-Elements.
- Use CFBundleShortVersionString to get version.
- Use notificationsTask.
- Add `@MainActor`.
- Use AsyncAlgorithms.

**Comment by tekezo** [comment]
# v15.3.36

-   Migrated to Swift 6.
-   As part of the Swift 6 migration, the internal processing of MultitouchExtension has been improved significantly.

**Comment by tekezo** [comment]
# v15.3.90

- Tweak UI of EventViewer.

**Comment by tekezo** [comment]
# v15.3.91

- Settings and Updater have also been built using Swift 6.

**Comment by tekezo** [comment]
# v15.3.92

- Fixed an issue where `iokit_hid_device_open_checker` was incorrectly reporting permission status, primarily due to Apple Headsets.

**Comment by tekezo** [comment]
# v15.3.93

- Fix publishing changes from within view updates in SettingsWindow/ContentMainView.swift.
- Use .listStyle(.sidebar) in DeviceSelectorView.
- Remove sidebarButtonStyle, sidebarButtonLabelStyle.

**Comment by tekezo** [comment]
# v15.3.94

- Fix to reflect changes in contentViewStates.navigationSelection in selectedSidebarItem.
